### PR TITLE
Fix the OS sensing code in ClusterFormationTasks

### PR DESCRIPTION
**Pull request by jbaiera**
This fixes a small bug in the sensing of the current OS family caused by Groovy
being too eager to cast non empty strings to the value `true`.

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
Lgtm. I have this fixed on the bundling jdk branch but hadn't yet ported to master because it happened to work with always using zip, for now. Thanks for fixing!

**Comment by jbaiera** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by jbaiera** [comment]
@elasticmachine run elasticsearch-ci/oss-distro-docs

**Comment by jbaiera** [comment]
@elasticmachine run elasticsearch-ci/1

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy [hunk 1]
```diff
@@ -350,14 +350,16 @@ class ClusterFormationTasks {
           the elasticsearch source tree then this should be the version of elasticsearch built by the source tree.
           If it isn't then Bad Things(TM) will happen. */
         Task extract = project.tasks.create(name: name, type: Copy, dependsOn: extractDependsOn) {
-            if (getOs().equals("windows")) {
-                from {
+            from {
+                File archiveFile = configuration.singleFile
+                if (archiveFile.getName().endsWith("zip")) {
                     project.zipTree(configuration.singleFile)
```
Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: Should this be using archiveFile?

File: docs/plugins/discovery-azure-classic.asciidoc [hunk 2]
```diff
@@ -388,6 +388,7 @@ This command should give you a JSON result:
 // TESTRESPONSE[s/"name" : "Cp8oag6",/"name" : "$body.name",/]
 // TESTRESPONSE[s/"cluster_name" : "elasticsearch",/"cluster_name" : "$body.cluster_name",/]
 // TESTRESPONSE[s/"cluster_uuid" : "AT69_T_DTp-1qgIJlatQqA",/"cluster_uuid" : "$body.cluster_uuid",/]
+// TESTRESPONSE[s/"build_type" : "zip",/"build_type" : "$body.version.build_type",/]
```
Comment (rjernst) [inline docs/plugins/discovery-azure-classic.asciidoc]: This can be done just like build flavor instead of manipulating the results.

**Review by rjernst — COMMENTED** [review]
A few comments

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy (hunk 1 above)
Comment (jbaiera) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: I just modified this section to be a direct porting of the change in your PR.

**Review by rjernst — COMMENTED** [review]
One last thing that was missing from my PR: we need the build type set in docs/Versions.asciidoc. Also I don't see the change to the docs test runner that passes in build type.

**Comment by atorok** [comment]
FYI I also did some of these Fixes today in #38899

**Review by alpar-t — APPROVED** [review]
LGTM

**Pull request merged**